### DISCUSSION TITLE
Algolia Integration - Sync Attributes to Products

### DISCUSSION THREAD

**Discussion by leonkosterss** [swellstores/community#171] [Ideas]
Hi,

It would be really nice if the Attributes added to a Swell Products will be synced to the Algolia `products` index.

We can use these fields to make Products filterable. I think this is a cleaner way then add custom Content fields.

For example I added these Attributes, but don't get them in Algolia.

<img width="1402" alt="image" src="https://user-images.githubusercontent.com/51200981/223678057-ae07856a-ae6f-4e2a-9bbd-cd6dade60c12.png">

<img width="1665" alt="image" src="https://user-images.githubusercontent.com/51200981/223678169-c9415008-5838-4bba-aa8a-38f2caffcfae.png">

Also, If i add new Content fields to a Product, these are not updating. I only get the new attributes when I delete the whole index, and save 1 product. Can this be updated, when I update the Product model? And can it sync all the products? Otherwise, I need to update all 1000 products, to get them in Algolia.

**Comment by Kealtie** [comment]
Vouch!

**Comment by ericingram** [comment]
This is a good idea.

To be transparent, we built a bunch of integrations with a limited scope in an effort to solve the most common use cases quickly. Unfortunately, we don't currently have the bandwidth to iterate on these integrations with our core team.

The solution (that we are working on now) is for Swell to introduce an app framework to enable anyone to build and maintain integrations among other things. Integrations we build in-house will be open-source, so you'll be able to add new functionality like this easily.

It will take time but stay tuned for announcements on our progress this year.
